DeepSpeech.ipynb

```diff
@@ -148,7 +148,16 @@
    },
    "outputs": [],
    "source": [
-    "dropout_rate = 0.05  # TODO: Validate this is a reasonable value"
+    "dropout_rate = 0.05  # TODO: Validate this is a reasonable value\n",
+    "\n",
+    "# This global placeholder will be used for all dropout definitions\n",
+    "dropout_rate_placeholder = tf.placeholder(tf.float32)\n",
+    "\n",
+    "# The feed_dict used for training employs the given dropout_rate\n",
+    "feed_dict_train = { dropout_rate_placeholder: dropout_rate }\n",
+    "\n",
+    "# While the feed_dict used for validation, test and train progress reporting employs zero dropout\n",
+    "feed_dict = { dropout_rate_placeholder: 0.0 }"
```

Comment:
This is a much cleaner way of doing things, and I think for small data sets, like the single scalar `dropout_rate`, it's likely fine.

However, one reason I moved away from `feed_dict` when going to the multi-gpu code is that  `feed_dict` is known to be slow[[1](https://github.com/tensorflow/tensorflow/issues/2919)]. In the thread [nanddalal](https://github.com/nanddalal) found using `feed_dict` with a GPU was about 4-5 times as slow as not using it on a GPU.

Quoting [yaroslavvb](https://github.com/yaroslavvb) (who was part of Google Brain and is part of OpenAI) from the thread referenced above:

> Feed_dict does a single-threaded memcpy of contents from Python runtime into TensorFlow runtime. If data is needed on GPU, then you'll have an additional CPU->GPU transfer. I'm used to seeing up to 10x improvement in performance when switching from feed_dict to native TensorFlow (Variable/Queue)

If we use `feed_dict`, we need to be careful not to put more stuff into `feed_dict`. (Which I guess is easiest if it's not there :-) )

You can make the call on keeping `feed_dict` or not. But just tell me your decision so I can start integrating my pull request in to yours before we merge yours.